src/hotspot/share/opto/doCall.cpp

```diff
@@ -247,7 +249,7 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
           // Look up second receiver.
           CallGenerator* next_hit_cg = NULL;
           ciMethod* next_receiver_method = NULL;
-          if (morphism == 2 && UseBimorphicInlining) {
+          if ((morphism == 2 && UseBimorphicInlining) || (morphism >= 2 && PolymorphicInlining)) {
             next_receiver_method = callee->resolve_invoke(jvms->method()->holder(),
                                                                profile.receiver(1));
             if (next_receiver_method != NULL) {
```

Comment:
Shall we disable inline for next_hit_cg if mophism > 2?

Reply:
Inline profile.receiver(1) has been tested to have little influence on performance. So maybe we can remove it here.